sos/report/plugins/__init__.py

```diff
@@ -2007,8 +2007,11 @@ def _add_cmd_output(self, **kwargs):
             kwargs['priority'] = 10
         if 'changes' not in kwargs:
             kwargs['changes'] = False
-        if self.get_option('all_logs') or kwargs['sizelimit'] == 0:
+        if (not getattr(SoSCommand(**kwargs), "snap_cmd", False) and
+           (self.get_option('all_logs') or kwargs['sizelimit'] == 0)):
```

Comment:
So the impact of the `snap_cmd` option will be "dont store directly to file even when all_logs is set"? That is independent on snap packaging, so I suggest changing option name.

E.g. why not to use `to_file` directly? We would just have to change condition at https://github.com/sosreport/sos/blob/main/sos/report/plugins/__init__.py#L2010 .

Reply:
we could use to_file directly to be set false for snap specific, but we can't afford to have `to_file` to be true if the command is being run is a snap command. We'd need a different test here for this then, if we drop the `snap_cmd` paramater. Ultimately `to_file` is False by default, so somehow we need to detect is the command being run is a snap or not, maybe check where the command is coming from, i.e. `/snap/bin`.

We don't want to do a blanket all commands to be not run to file, as journalctl wouldn't be snapped, and hence can write to file.

So, we could have for example

sunbeam cluster list <--snap
journalctl <-- not snap

now, the latter command, we can run with `to_file=True`, but the former we can't

Reply:
I see. One part of my concern was also to the parameter name itself, trying to keep `Plugin` independent on packaging implementation (same concernt I would raise against, say, `is_rpm` parameter name). But I cant figure out some generic enough name (`force_to_file`? too long..) that I would suggest to use.

So ACK from me.